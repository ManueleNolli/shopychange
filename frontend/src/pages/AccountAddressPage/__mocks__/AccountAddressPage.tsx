import { divWithoutChildrenMock } from '../../../../mocksUtils'

const AccountAddressPageMock = () =>
  divWithoutChildrenMock('AccountAddressPage')
export default AccountAddressPageMock
