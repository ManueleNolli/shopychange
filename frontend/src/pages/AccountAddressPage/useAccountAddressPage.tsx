import { useEffect, useState } from 'react'
import { useUserContext } from '../../context/userContext'
import useFetching from '../../hooks/useFetching'
import { NFT } from '../../types/components/NFT'
import {
  handlerQueryErrorCatch,
  ownedNftsQuery,
} from '../../services/BackendService'

export default function useAccountAddressPage(address: string) {
  const { blockchainNetworkId } = useUserContext()
  const { isFetching, setIsFetching, isError, setIsError } = useFetching()
  const [nfts, setNfts] = useState<NFT[]>([])

  useEffect(() => {
    const fetchNFTs = async () => {
      if (!blockchainNetworkId) return

      const response = await handlerQueryErrorCatch(
        ownedNftsQuery(blockchainNetworkId, address)
      )

      if (response.error) {
        setIsError(true)
        setIsFetching(false)
        return
      }

      const nfts = response.result.ownedNfts as NFT[]
      setNfts(nfts)
      setIsFetching(false)
    }

    fetchNFTs()
  })

  return {
    isFetching,
    isError,
    nfts,
  }
}
