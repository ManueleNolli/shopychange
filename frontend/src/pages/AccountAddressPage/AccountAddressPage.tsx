import React from 'react'

import { Box, Center, Heading } from '@chakra-ui/react'

import { AccountAddressPageProps } from '../../types/props/AccountAddressPageProps'
import { simplifiedAddress } from '../../utils/SimplifiedAddress/SimplifiedAddress'
import useAccountAddressPage from './useAccountAddressPage'
import QueryContainer from '../../utils/QueryContainer/QueryContainer'
import NFTOrCollectionViewer from '../../components/NFTOrCollectionViewer/NFTOrCollectionViewer'

export default function AccountAddressPage({
  address,
}: AccountAddressPageProps) {
  const { isFetching, isError, nfts } = useAccountAddressPage(address)

  return (
    <Box>
      <Center>
        <Heading>Account {simplifiedAddress(address)}</Heading>
      </Center>

      <QueryContainer isLoading={isFetching} isError={isError}>
        <Center>
          <NFTOrCollectionViewer nfts={nfts} />
        </Center>
      </QueryContainer>
    </Box>
  )
}
