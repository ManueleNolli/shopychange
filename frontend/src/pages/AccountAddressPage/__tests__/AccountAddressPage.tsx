import React from 'react'
import { render } from '@testing-library/react'
import AccountAddressPage from '../AccountAddressPage'
import useAccountAddressPage from '../useAccountAddressPage'

jest.mock('../../../components/NFTOrCollectionViewer/NFTOrCollectionViewer')
jest.mock('../useAccountAddressPage', () => jest.fn())
const mockProps = {
  address: '0x1203A6e8bD6A063AEa9D4B7F8A2BaE9405fE44Cd',
}
jest.mock('../../../utils/SimplifiedAddress/SimplifiedAddress', () => ({
  simplifiedAddress: jest.fn((address: string) => address),
}))

const mockedData = {
  ownedNfts: [
    {
      contractAddress: '0x0000000000',
      tokenId: '0',
      image: 'https://example.com/image.png',
      name: 'Test NFT',
    },
    {
      contractAddress: '0x0000000001',
      tokenId: '1',
      image: 'https://example.com/image.png',
      name: 'Test1 NFT',
    },
  ],
}

describe('AccountAddressPage', () => {
  test('renders successfully', () => {
    ;(useAccountAddressPage as jest.Mock).mockReturnValue({
      isFetching: false,
      isError: false,
      nfts: mockedData.ownedNfts,
    })

    const tree = render(<AccountAddressPage {...mockProps} />)
    expect(tree.container).toMatchSnapshot()
  })
})
