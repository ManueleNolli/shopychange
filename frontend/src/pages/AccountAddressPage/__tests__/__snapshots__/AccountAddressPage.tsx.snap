// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountAddressPage renders successfully 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-gmuwbf"
    >
      <h2
        class="chakra-heading css-0"
      >
        Account 
      </h2>
    </div>
    <div
      class="css-gmuwbf"
    >
      <div
        data-testid="NFTOrCollectionViewer"
      />
    </div>
  </div>
</div>
`;
